src/test/java/org/opensearch/geospatial/ip2geo/Ip2GeoDataServer.java

```diff
@@ -0,0 +1,120 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.common.SuppressForbidden;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+/**
+ * Simple http server to serve static files under test/java/resources/ip2geo/server for integration testing
+ */
+@Log4j2
+@SuppressForbidden(reason = "used only for testing")
+public class Ip2GeoDataServer {
+    private static final String SYS_PROPERTY_KEY_CLUSTER_ENDPOINT = "tests.rest.cluster";
+    private static final String LOCAL_CLUSTER_ENDPOINT = "127.0.0.1";
+    private static final String ROOT = "ip2geo/server";
+    private static final int PORT = 8001;
+
+    private static HttpServer server;
+    private static volatile int counter = 0;
+    private static String endpointPrefix = "http://localhost:" + PORT;
+    private static String cityFilePath = endpointPrefix + "/city/manifest_local.json";
+    private static String countryFilePath = endpointPrefix + "/country/manifest_local.json";
+
+    /**
+     * Return an endpoint to a manifest file for a sample city data
+     * The sample data should contain three lines as follows
+     *
+     * cidr,city,country
+     * 10.0.0.0/8,Seattle,USA
+     * 127.0.0.0/12,Vancouver,Canada
+     * fd12:2345:6789:1::/64,Bengaluru,India
+     *
+     */
+    public static String getEndpointCity() {
+        return cityFilePath;
+    }
+
+    /**
+     * Return an endpoint to a manifest file for a sample country data
+     * The sample data should contain three lines as follows
+     *
+     * cidr,country
+     * 10.0.0.0/8,USA
+     * 127.0.0.0/12,Canada
+     * fd12:2345:6789:1::/64,India
+     *
+     */
+    public static String getEndpointCountry() {
+        return countryFilePath;
+    }
+
+    @SneakyThrows
+    synchronized public static void start() {
+        // If it is remote cluster test, use external endpoint and do not launch local server
+        if (System.getProperty(SYS_PROPERTY_KEY_CLUSTER_ENDPOINT).contains(LOCAL_CLUSTER_ENDPOINT) == false) {
+            String externalEndpointPrefix = "https://github.com/opensearch-project/geospatial/blob/main/src/test/resources/ip2geo/server";
+            cityFilePath = externalEndpointPrefix + "/city/manifest.json";
+            countryFilePath = externalEndpointPrefix + "/country/manifest.json";
+            return;
+        }
+
+        counter++;
+        if (server != null) {
+            return;
+        }
+        server = HttpServer.create(new InetSocketAddress("localhost", PORT), 0);
+        server.createContext("/", new Ip2GeoHttpHandler());
+        server.start();
+        log.info("Local file server started on port {}", PORT);
```

Comment:
can we add similar log for before start event? in case server failed to start such log message can help in troubleshooting